internal/config/options.go

```diff
@@ -188,17 +203,17 @@ func NewOptions(authenticateURL, authorizeURL string) (*Options, error) {
 // options by parsing environmental variables and config file
 func OptionsFromViper(configFile string) (*Options, error) {
 	// start a copy of the default options
-	o := defaultOptions
+	o := NewDefaultOptions()
 	// Load up config
 	o.bindEnvs()
 	if configFile != "" {
-		viper.SetConfigFile(configFile)
-		if err := viper.ReadInConfig(); err != nil {
+		o.viper.SetConfigFile(configFile)
+		if err := o.viper.ReadInConfig(); err != nil {
 			return nil, fmt.Errorf("internal/config: failed to read config: %s", err)
 		}
 	}
 
-	if err := viper.Unmarshal(&o); err != nil {
+	if err := o.viper.Unmarshal(&o); err != nil {
```

Comment:
This is a bit weird but works in practice.  An alternative would be a wrapper struct but that doesn't feel worth it.  We could also make a new instance and then place it into the struct after Unmarshall().

Reply:
That's pretty cool this works. 
That said, for clarity and readability, (and unintended state from the original?) I think it might be worth a bit of verbosity to create a new uninitialized struct like you are saying. But I don't feel strongly.

Reply:
bindEnvs() gets in the way a bit but here's an attempt https://github.com/pomerium/pomerium/pull/332/commits/60e64f971c5e9ea10cadc677dd9936303fca6208

I don't hate it :)

Reply:
It's fine. 👍 


🤣